proctor-common/src/main/java/com/indeed/proctor/common/RuleEvaluator.java

```diff
@@ -229,6 +231,26 @@ static void checkRuleIsBooleanType(
         // To support users writing rules, be more strict here in requiring the type of the
         // value to be expected before coercion
         Class<?> type = ve.getType(elContext);
+
+        // if object is map check what the base value type is
+        if (Objects.equals(type, Object.class)) {
```

Comment:
i use debugger and see the type is always Boolean when executing the test case testElExpressionsShouldBeAvailableNestedMap.

In what condition the type will be Object?

Reply:
For the first case: `${foo.bar == true}`, it will show Boolean as it recognizes the boolean comparison. 

In the second case `${foo.bar}` it should show as an Object.

Reply:
The first case was passing already but was just a test case missing in general with using a map.